pkgs/shelf_static/lib/src/static_handler.dart

```diff
@@ -39,6 +39,11 @@ final _defaultMimeTypeResolver = MimeTypeResolver();
 ///
 /// Specify a custom [contentTypeResolver] to customize automatic content type
 /// detection.
+///
+/// The [Response.context] will be populated with "shelf_static:file" or
+/// "shelf_static:file_not_found" with the resolved [File] for the [Response],
+/// while respecting [serveFilesOutsidePath] to prevent exposing a [File] path
```

Comment:
Call out the specific behavior for `serveFilesOutsidePath`. Maybe something like

"If the file is considered not found because it is outside of the [fileSystemPath] and [serveFilesOutsidePath] is false then neither key will be included in the context."

Reply:
How about?:

```markdown
/// The [Response.context] will be populated with "shelf_static:file" or
/// "shelf_static:file_not_found" with the resolved [File] for the [Response].
/// If the file is considered not found because it is outside of the
/// [fileSystemPath] and [serveFilesOutsidePath] is false, then neither key
/// will be included in the context.
```

- If you confirm this version I will change the text in the code.

Reply:
> How about?:

Looks Good To Me